ci/cibw_before_all_macos.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+set -eo pipefail
+
+if [[ "$1" == "" ]] ; then
+    echo "Usage: $0 <PROJECT_PATH>"
+    exit 1
+fi
+PROJECT_PATH="$1"
+ARCH=$(uname -m)
+export HDF5_VERSION="1.12.2"
+export HDF5_DIR="$PROJECT_PATH/cache/hdf5/$HDF5_VERSION-$ARCH"
+# When compiling HDF5, we should use the minimum across all Python versions for a given
+# arch, for versions see for example a more updated version of the following:
+# https://github.com/pypa/cibuildwheel/blob/89a5cfe2721c179f4368a2790669e697759b6644/cibuildwheel/macos.py#L296-L310
+if [[ "$ARCH" == "arm64" ]]; then
+    export MACOSX_DEPLOYMENT_TARGET="11.0"
+else
+    export MACOSX_DEPLOYMENT_TARGET="10.9"
+fi
```

Comment:
Did not seem to work:
```
  delocate.libsana.DelocationError: Library dependencies do not satisfy target MacOS version 11.0:
  /private/var/folders/xz/lz5thm6s3vb1vdkk77vmkgbr0000gn/T/tmpxrqwmqvd/wheel/h5py/.dylibs/libhdf5.200.dylib has a minimum target of 14.0
```

Reply:
Not a big deal. Thanks for trying anyway !

Reply:
... and now I remember why. In principle this should be a `before_all` step, but in order to cache it properly I thought I needed to run it separately. But I don't think I need to actually... I'll try rolling it into regular cibuildwheel `before_all` and see if it works.

Reply:
Okay I think the sticking point here is that we need to set a bunch of env vars when getting HDF5. On macOS it's LD_LIBRARY_PATH, PKG_CONFIG_PATH, etc. On Windows it's CL, LINK, ZLIB_ROOT, etc. If we do these steps *before* cibuildwheel is invoked it's easy -- we just `echo "ZLIB_ROOT=$ZLIB_ROOT" | tee -a $GITHUB_ENV` for example.

If we're *already in the cibuildwheel step* this does not work -- it doesn't add the env vars to the currently running step's (cibuildwheel step's) environment. And I don't think you can `source` the scripts in the `before_all` to get env vars back to the cibuildwheel step AFAIK.

One way around this would be to define `CIBUILDWHEEL_ENVIRONMENT_MACOS` for example in `pyproject.toml` or in `build_wheels.yml` to set the vars, but I think this is overall a step back -- it pushes setting their values to a YAML or TOML file whereas currently they're set in a script where they're defined.

So I'm going to revert things a bit and fix the macOS build, and mention why it's done this way.

Reply:
excellent, thank you !